### PR TITLE
[1LP][RFR] REST API tests for custom attributes refactoring and new tests

### PR DESCRIPTION
**Pull request by mkoura**
* refactoring existing REST API custom attribute tests (these were one of my first tests and I don't like them anymore :)
* adding new tests for testing adding "section" with bad value

{{pytest: '''cfme/tests/infrastructure/test_providers.py::TestProvidersRESTAPI'' -v'}}

### CODE REVIEW

File: cfme/tests/infrastructure/test_providers.py [hunk 1]
```diff
@@ -250,24 +250,38 @@ class TestProvidersRESTAPI(object):
...
+            body.append({
+                'name': 'ca_name_{}'.format(uid),
+                'value': 'ca_value_{}'.format(uid)
+            })
+        attrs = provider.custom_attributes.action.add(*body)
 
         yield attrs
 
         try:
+            # custom attributes can be deleted by tests, just log warning
```
Comment (mshriver) [inline cfme/tests/infrastructure/test_providers.py]: did you mean to log here?

File: cfme/tests/infrastructure/test_providers.py [hunk 2]
```diff
@@ -250,24 +250,38 @@ class TestProvidersRESTAPI(object):
...
     @test_requirements.rest
-    def test_add_custom_attributes(self, custom_attributes):
-        """Test adding custom attributes to provider using REST API."""
+    def test_add_custom_attributes(self, rest_api, custom_attributes):
+        """Test adding custom attributes to provider using REST API.
+
+        Metadata:
+            test_flag: rest
+        """
         assert len(custom_attributes) == 2
```
Comment (mshriver) [inline cfme/tests/infrastructure/test_providers.py]: I think you should drop this check, or not use the literal '2' but a class constant here and in the `custom_attributes` fixture

File: cfme/tests/infrastructure/test_providers.py [hunk 3]
```diff
@@ -250,24 +250,38 @@ class TestProvidersRESTAPI(object):
...
-        assert custom_attributes[0].name == "ca_name1" and custom_attributes[0].value == "ca_value1"
-        assert custom_attributes[1].name == "ca_name2" and custom_attributes[1].value == "ca_value2"
+        provider = rest_api.collections.providers.get(id=custom_attributes[0].resource_id)
```
Comment (mshriver) [inline cfme/tests/infrastructure/test_providers.py]: Consider yielding both the attr and the provider from `custom_attributes`

**Comment by mkoura** [comment]
@mshriver thanks for review and suggestions

File: cfme/tests/infrastructure/test_providers.py [hunk 4]
```diff
@@ -295,17 +315,77 @@ def test_delete_custom_attributes(self, rest_api, custom_attributes, from_detail
...
+    @pytest.mark.tier(3)
+    @test_requirements.rest
+    def test_add_custom_attributes_bad_section(self, rest_api, setup_a_provider):
+        """Test that adding custom attributes with invalid section
+        to provider using REST API fails.
+
+        Metadata:
+            test_flag: rest
+        """
+        provider = rest_api.collections.providers[0]
```
Comment (psav) [inline cfme/tests/infrastructure/test_providers.py]: Couldn't we get this from current appliance? I think this is the preferred way now

File: cfme/tests/infrastructure/test_providers.py [hunk 5]
```diff
@@ -249,63 +250,142 @@ def test_provider_crud(provider):
...
+    @pytest.mark.tier(3)
+    @test_requirements.rest
+    def test_add_custom_attributes_bad_section(self, rest_api, setup_a_provider):
+        """Test that adding custom attributes with invalid section
+        to provider using REST API fails.
+
+        Metadata:
+            test_flag: rest
+        """
+        provider = rest_api.collections.providers.get(name=setup_a_provider.name)
```
Comment (mkoura) [inline cfme/tests/infrastructure/test_providers.py]: @psav I changed it to this to make sure that I use the provider returned by the fixture.

Comment (psav) [inline cfme/tests/infrastructure/test_providers.py]: ok

File: cfme/tests/infrastructure/test_providers.py [hunk 6]
```diff
@@ -249,63 +250,142 @@ def test_provider_crud(provider):
 class TestProvidersRESTAPI(object):
     @pytest.yield_fixture(scope="function")
     def custom_attributes(self, rest_api, setup_a_provider):
-        provider = rest_api.collections.providers[0]
-        attrs = provider.custom_attributes.action.add(
-            {"name": "ca_name1", "value": "ca_value1"}, {"name": "ca_name2", "value": "ca_value2"})
-
-        yield attrs
+        provider = rest_api.collections.providers.get(name=setup_a_provider.name)
```
Comment (mkoura) [inline cfme/tests/infrastructure/test_providers.py]: It's not the provider object, it's a the /api/providers/:id resource that corresponds to the provider object returned by `setup_a_provider` fixture.

**Comment by mkoura** [comment]
@psav @mfalesni @jkrocil please review

**Pull request merged**